### PR TITLE
Adding ndarray_to_rdd function

### PR DESCRIPTION
**Pull request by gaarsmu**
This is my first pull request, so don't be too angry...

## What changes were proposed in this pull request?

Adding a function, which converts numpy ndarray to rdd.

## How was this patch tested?
I used tested this function in gradient descent algorithm realization, and tested Errors, could be find here:
https://github.com/gaarsmu/BigDL_notebooks/tree/master/to_RDD_func

### CODE REVIEW

File: pyspark/bigdl/util/common.py [hunk 1]
```diff
@@ -434,6 +434,18 @@ def _test():
     if failure_count:
         exit(-1)
 
+class TransformationError(BaseException):
+    pass
+
+def ndarray_to_rdd(X, y):
```
Comment (zhichao-li) [inline pyspark/bigdl/util/common.py]: How about `to_sample_rdd` or `to_rdd_of_sample` ?

File: pyspark/bigdl/util/common.py [hunk 2]
```diff
@@ -434,6 +434,18 @@ def _test():
...
+    if type(X) != np.ndarray or type(y) != np.ndarray:
+        raise TransformationError('Input arrays should be numpy ndarrays')
```
Comment (zhichao-li) [inline pyspark/bigdl/util/common.py]: I think simply use `Exception` is fine here as we haven't have uniform customized exception for now.

File: pyspark/bigdl/util/common.py (hunk 1 above)
Comment (zhichao-li) [inline pyspark/bigdl/util/common.py]: add another parameter: `numSlices=None` to control the partition num?

Comment (gaarsmu) [reply pyspark/bigdl/util/common.py]: I've fixed two issues you've mentioned, but got a question about this one:
Is this partition number a pre-build function of Spark Contex, of you want to to simple add it up manually? (like slice ndarrays to partitions and parallelize each of them)

**Comment by zhichao-li** [comment]
there's a similar function: `to_sample_rdd` to fix this.

**Pull request closed without merging**